app/assets/javascripts/travelling_game/fire_ball.js

```diff
@@ -0,0 +1,67 @@
+var FireBall = function(origin_x, origin_y, target_x, target_y, damage){
+  this.origin_x = origin_x;
+  this.origin_y = origin_y;
+  this.target_x = target_x;
+  this.target_y = target_y;
+  this.x = origin_x;
+  this.y = origin_y;
+  this.damage = damage;
+  this.total_distance = Math.sqrt(Math.pow(target_x - origin_x,2) + Math.pow(target_y - origin_y,2))
+  this.v_x = 3*(target_x - origin_x)/this.total_distance
+  this.v_y = 3*(target_y - origin_y)/this.total_distance
+}
+
+FireBall.prototype.distance_travelled = function(){
+  return Math.sqrt( Math.pow(this.x - this.origin_x,2) + Math.pow(this.y - this.origin_y,2) )
```

Comment:
Line is too long.<br>Identifier 'origin_x' is not in camel case.<br>Identifier 'origin_y' is not in camel case.<br>Missing semicolon.